test/example/module-splitting.txt

```diff
@@ -0,0 +1,448 @@
+Before:
+(module
+)
+After:
+(module
+)
+Secondary:
+(module
+)
+
+
+Before:
+(module
+ (type $i32_=>_none (func (param i32)))
+ (memory $mem (shared 3 42))
+ (table $table 3 42 funcref)
+ (global $glob (mut i32) (i32.const 7))
+ (event $e (attr 0) (param i32))
+)
+After:
+(module
+ (type $i32_=>_none (func (param i32)))
+ (memory $mem (shared 3 42))
+ (table $table 3 42 funcref)
+ (global $glob (mut i32) (i32.const 7))
+ (event $e (attr 0) (param i32))
+ (export "memory" (memory $mem))
+ (export "table" (table $table))
+ (export "global" (global $glob))
+ (export "event" (event $e))
+)
+Secondary:
+(module
+ (type $i32_=>_none (func (param i32)))
+ (import "primary" "memory" (memory $mem (shared 3 42)))
+ (import "primary" "table" (table $table 3 42 funcref))
+ (import "primary" "global" (global $glob (mut i32)))
+ (import "primary" "event" (event $e (attr 0) (param i32)))
+)
+
+
+Before:
+(module
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (func $foo (param $0 i32) (result i32)
+  (local.get $0)
+ )
+)
+After:
+(module
+)
+Secondary:
+(module
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (func $foo (param $0 i32) (result i32)
+  (local.get $0)
+ )
+)
+
+
+Before:
+(module
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (export "foo" (func $foo))
+ (func $foo (param $0 i32) (result i32)
+  (local.get $0)
+ )
+)
+After:
+(module
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (import "placeholder" "0" (func $placeholder_0 (param i32) (result i32)))
+ (table $0 1 funcref)
+ (elem (i32.const 0) $placeholder_0)
+ (export "foo" (func $foo))
+ (func $foo (param $0 i32) (result i32)
+  (call_indirect (type $i32_=>_i32)
+   (local.get $0)
+   (i32.const 0)
+  )
+ )
+)
+Secondary:
+(module
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (table $0 1 funcref)
```

Comment:
In case the primary module doesn't originally have a table but gets one during the module splitting process, the table is not exported, and it seems here the primary and the secondary module don't share the table and they end up with separate tables. Then it looks, how is the secondary module supposed to replace the table entries?

Reply:
Thanks for catching this. This is a bug that I missed in my first readthrough of the test output. The problem was that the table was created after the code to share it had already run. The fix is to move `shareImportableItems` to run last rather than first.